include/spdlog/sinks/ansicolor_sink.h

```diff
@@ -138,6 +139,11 @@ class ansicolor_sink final : public sink
         return should_do_colors_;
     }
 
+    void force_color_output(bool value)
+    {
+        should_do_colors_ = value || can_do_colors_;
```

Comment:
A bit confusing. It is called force so should be forced.  I think can_do_colors_ is not needed. just set here should_do_colors_ = value

Reply:
The idea here is that if at some point the user decides they no longer want to force color output, the sink should return to its default behavior, i.e., printing color only if a TTY is found. So if the user calls `force_color_output(false)` the variable `should_do_colors` still remains true if there is a TTY with color support. Otherwise it doesn't make sense to have a `value` parameter in the first place (or the function should be called something like `enable_color_output`). I think this interface is more flexible and properly reflects the semantics of "forcing" color output.

Reply:
but how would one turn off the color?

Reply:
Ah, I see. We obviously had different behaviors in mind for this method. I just assumed if you don't want color output in the first place, you'd use a different sink. The problem I see with the semantics you suggest is that once this function has been called with either `true` or `false`, there is no way to get back to the sink's original "automatic" behavior. It's certainly interesting, as to me "forcing" means that I definitely want a certain behavior, while "not forcing" means that I don't really care (i.e., let the system decide for me).

I'm fine with either interpretation, however if we go with yours I'd suggest to rename it to `enable_color_output`, as in my mind it more clearly reflects what is happening. Also in that case the added parameter for the factory functions wouldn't work, as it would always either enable or disable color output, regardless of TTY or not. I guess we'd then have to provide overloads instead.

Reply:
Or pass in an enum value, one of `ALWAYS_COLOR`, `NEVER_COLOR`, or `AUTO_COLOR`. This would allow you to force color output, force no color output, or auto-detect whether there should be color output, respectively.

Best of both worlds!

Reply:
Yes, I'd like that solution as well!

Reply:
Thanks @godbyk - thats indeed good and clear solution. Might even be better to pass this in enum to the ctor after all. This solution even enable me to remove the redundant (non color) stdout/stderr sinks in future.